Sources/Basic/PathShims.swift

```diff
@@ -44,11 +44,11 @@ public func exists(_ path: AbsolutePath, followSymlink: Bool = true) -> Bool {
 }
 
 /// Returns true if and only if `path` refers to an existent file system entity and that entity is a regular file.
-/// If `followSymlink` is true, and the last path component is a symbolic link, the result pertains to the destination 
+/// If `followSymlink` is true, and the last path component is a symbolic link, the result pertains to the destination
 /// of the symlink; otherwise it pertains to the symlink itself. If any file system error other than non-existence
 /// occurs, this function throws an error.
 public func isFile(_ path: AbsolutePath, followSymlink: Bool = true) -> Bool {
-    guard let status = try? stat(path, followSymlink: followSymlink), status.kind == .file else {
+    guard let status = try? stat(path, followSymlink: followSymlink), status.st_mode & S_IFREG != 0 else {
```

Comment:
Does this make the extension in POSIX dead code? If yes, should we just remove it? Maybe we can convert the lower layer into a struct of some kind so there is no namespacing issues. I am assuming you filed a bug for the compiler folks already.

Reply:
Well the problem is that for whatever reason, the module in the BASIC isn't picking up the extension in POSIX. Ideally we'd be able to use that as a single source of truth, but it might require a compiler bug fix, or alternatively, a modification to the POSIX library to use its own struct wrapping around the st_mode (which would be a zero-cost abstraction and fix the problem). I'm not sure whether that would be a better or worse way of solving the issue.